modules/install-couchbase-server/install-couchbase-server

```diff
@@ -110,10 +110,11 @@ function install_couchbase_on_amazon_linux {
   local readonly version="$2"
   local readonly checksum="$3"
   local readonly checksum_type="$4"
+  local readonly base_os_version="$5"
 
   log_info "Installing Couchbase $version ($edition edition) on Amazon Linux"
 
-  local readonly filepath="couchbase-server-${edition}-${version}-centos7.x86_64.rpm"
+  local readonly filepath="couchbase-server-${edition}-${version}-${base_os_version}.x86_64.rpm"
```

Comment:
Is this fix still needed in this PR?

Reply:
Yes, that parameter is passed through based on what version of CB you're installing. 

```bash
[[ $version > 5.1.1 ]] && base_os_version="amzn2" || base_os_version="centos6"
    install_couchbase_on_amazon_linux "$edition" "$version" "$checksum" "$checksum_type" "$base_os_version"

```

Reply:
In this case, it will fallback to `centos6` because that version is compatible with "Amazon Linux". Despite the Centos6 version won't appear in the website's download list, it still there in their servers.

So, if the user chose to install CB =< 5.1.1 using "Amazon Linux", centos6 is the right choice.

By the way, my team is using this config for our production cluster.

Reply:
Hm, I'm not sure that'll work, as the different version would have different checksums, so the install will fail. Also, we no longer test on Amazon Linux 1, only Amazon Linux 2, so I don't think introducing this change makes sense.

Reply:
How would you keep retro compatibility with CB 5.1.1 and lower since they are using Centos6 only?
This change is setting Amazon Linux 2 as the default option for CB 6 (or higher) instead of Centos7.

By the way, I'm using the fork in my production pipeline and it's working like a charm.

Reply:
The latest version of this repo only supports AL2. 

Also, I'm super happy to hear it's working for you in prod. That said, I'd like to understand _why_ it's working. I'd expect the checksums validation to fail with the tweak in this PR. Unless, of course, under the hood, CB is using the exact same binaries, but that seems unlikely. Perhaps add some logging and see what's going on?

Reply:
Fair enough. I will close this PR and keep using the fork version then.
At this point, seems this project will drop the support for Couchbase < 5.1.1, so I'd recommend you guys to update the documentation explaining that.